Traceback (most recent call last):
  File "c:\users\yamta\miniconda3\lib\site-packages\nbclient\client.py", line 599, in _async_poll_for_reply
    msg = await ensure_async(self.kc.shell_channel.get_msg(timeout=new_timeout))
  File "c:\users\yamta\miniconda3\lib\site-packages\nbclient\util.py", line 85, in ensure_async
    result = await obj
  File "c:\users\yamta\miniconda3\lib\site-packages\jupyter_client\asynchronous\channels.py", line 48, in get_msg
    raise Empty
_queue.Empty

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\users\yamta\miniconda3\lib\site-packages\jupyter_cache\executors\utils.py", line 56, in single_nb_execution
    record_timing=False,
  File "c:\users\yamta\miniconda3\lib\site-packages\nbclient\client.py", line 1077, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "c:\users\yamta\miniconda3\lib\site-packages\nbclient\util.py", line 74, in wrapped
    return just_run(coro(*args, **kwargs))
  File "c:\users\yamta\miniconda3\lib\site-packages\nbclient\util.py", line 53, in just_run
    return loop.run_until_complete(coro)
  File "c:\users\yamta\miniconda3\lib\asyncio\base_events.py", line 583, in run_until_complete
    return future.result()
  File "c:\users\yamta\miniconda3\lib\site-packages\nbclient\client.py", line 527, in async_execute
    cell, index, execution_count=self.code_cells_executed + 1
  File "c:\users\yamta\miniconda3\lib\site-packages\nbclient\client.py", line 806, in async_execute_cell
    exec_reply = await self.task_poll_for_reply
  File "c:\users\yamta\miniconda3\lib\site-packages\nbclient\client.py", line 623, in _async_poll_for_reply
    await self._async_handle_timeout(timeout, cell)
  File "c:\users\yamta\miniconda3\lib\site-packages\nbclient\client.py", line 674, in _async_handle_timeout
    "Cell execution timed out", timeout, cell
nbclient.exceptions.CellTimeoutError: A cell timed out while it was being executed, after 30 seconds.
The message was: Cell execution timed out.
Here is a preview of the cell contents:
-------------------
# Simulation constants
num_iter = 500 # number of iterations
nt_max = 1000 # Maximum number of simulation time
batch_size = 250 # Batch size

sz = 16 # image patch size
num_units = 100 # number of neurons (units)
eps = 1e-2 # small value which determines convergence

model, errorarr = run_simulation(imgs, num_iter, nt_max, batch_size, sz, num_units, eps)
-------------------

